# Setup thư viện

In [ ]:
!pip install ultralytics

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 950.0/950.0 kB 32.5 MB/s eta 0:00:00


In [ ]:
!yolo --version

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.
WARNING ⚠️ argument '--version' does not require leading dashes '--', updating to 'version'.
8.3.97


# Predict với pretrain weights

In [ ]:
from ultralytics import YOLO

# Đường dẫn lưu model
save_path = "/kaggle/working/yolov8m.pt"

# Tải model nếu chưa có
model = YOLO("yolov8m.pt")
print("Tải model thành công:", model)

# Chạy inference với hình ảnh online
model.predict(source="https://ultralytics.com/images/bus.jpg", save=True)

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


100%|██████████| 49.7M/49.7M [00:00<00:00, 111MB/s] 


Tải model thành công: YOLO(
  (model): DetectionModel(
    (model): Sequential(
      (0): Conv(
        (conv): Conv2d(3, 48, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(48, eps=0.001, momentum=0.03, affine=True, track_running_stats=True)
        (act): SiLU(inplace=True)
      )
      (1): Conv(
        (conv): Conv2d(48, 96, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(96, eps=0.001, momentum=0.03, affine=True, track_running_stats=True)
        (act): SiLU(inplace=True)
      )
      (2): C2f(
        (cv1): Conv(
          (conv): Conv2d(96, 96, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(96, eps=0.001, momentum=0.03, affine=True, track_running_stats=True)
          (act): SiLU(inplace=True)
        )
        (cv2): Conv(
          (conv): Conv2d(192, 96, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(96, eps=0.001, momentum=0.03, affin

100%|██████████| 134k/134k [00:00<00:00, 17.0MB/s]


image 1/1 /kaggle/working/bus.jpg: 640x480 4 persons, 1 bus, 41.5ms
Speed: 21.7ms preprocess, 41.5ms inference, 253.3ms postprocess per image at shape (1, 3, 640, 480)
Results saved to runs/detect/predict


[ultralytics.engine.results.Results object with attributes:
 
 boxes: ultralytics.engine.results.Boxes object
 keypoints: None
 masks: None
 names: {0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', 10: 'fire hydrant', 11: 'stop sign', 12: 'parking meter', 13: 'bench', 14: 'bird', 15: 'cat', 16: 'dog', 17: 'horse', 18: 'sheep', 19: 'cow', 20: 'elephant', 21: 'bear', 22: 'zebra', 23: 'giraffe', 24: 'backpack', 25: 'umbrella', 26: 'handbag', 27: 'tie', 28: 'suitcase', 29: 'frisbee', 30: 'skis', 31: 'snowboard', 32: 'sports ball', 33: 'kite', 34: 'baseball bat', 35: 'baseball glove', 36: 'skateboard', 37: 'surfboard', 38: 'tennis racket', 39: 'bottle', 40: 'wine glass', 41: 'cup', 42: 'fork', 43: 'knife', 44: 'spoon', 45: 'bowl', 46: 'banana', 47: 'apple', 48: 'sandwich', 49: 'orange', 50: 'broccoli', 51: 'carrot', 52: 'hot dog', 53: 'pizza', 54: 'donut', 55: 'cake', 56: 'chair', 57: 'couch', 58: 'potted p

In [ ]:
# Predict detection bằng CLI
!yolo task=detect mode=predict model=yolov8m.pt source="https://nextcity.org/images/made/219951734_2838e034bb_o_840_630_80.jpg"

Ultralytics 8.3.98 🚀 Python-3.10.12 torch-2.5.1+cu121 CUDA:0 (Tesla P100-PCIE-16GB, 16269MiB)
YOLOv8m summary (fused): 92 layers, 25,886,080 parameters, 0 gradients, 78.9 GFLOPs

100%|████████████████████████████████████████| 159k/159k [00:00<00:00, 18.9MB/s]
image 1/1 /kaggle/working/219951734_2838e034bb_o_840_630_80.jpg: 480x640 15 persons, 4 bicycles, 4 motorcycles, 39.8ms
Speed: 6.2ms preprocess, 39.8ms inference, 176.6ms postprocess per image at shape (1, 3, 480, 640)
Results saved to runs/detect/predict2
💡 Learn more at https://docs.ultralytics.com/modes/predict


In [ ]:
from ultralytics import YOLO
import cv2
import matplotlib.pyplot as plt

# Load model YOLOv8m-seg
model = YOLO("yolov8m.pt")

# Predict trên ảnh từ URL (tải ảnh xuống)
image_url = "https://nextcity.org/images/made/219951734_2838e034bb_o_840_630_80.jpg"
results = model.predict(source=image_url, save=True)



Found https://nextcity.org/images/made/219951734_2838e034bb_o_840_630_80.jpg locally at 219951734_2838e034bb_o_840_630_80.jpg
image 1/1 /kaggle/working/219951734_2838e034bb_o_840_630_80.jpg: 480x640 15 persons, 4 bicycles, 4 motorcycles, 40.2ms
Speed: 2.3ms preprocess, 40.2ms inference, 1.7ms postprocess per image at shape (1, 3, 480, 640)
Results saved to runs/detect/predict3


In [ ]:
ls

219951734_2838e034bb_o_840_630_80.jpg  bus.jpg  runs/  yolov8m.pt


## Training YOLO v8m model with Custom data

In [ ]:
import os

# Đường dẫn lưu dataset
save_dir = "/kaggle/working/data"

# Link tải dataset từ Roboflow
dataset_url = "https://app.roboflow.com/ds/C9Yd3UmR0F?key=3lwjSdzdA8"

# Lệnh tải file
os.system(f"wget -O {save_dir}/dataset.zip '{dataset_url}'")
os.system(f"unzip {save_dir}/dataset.zip -d {save_dir}")


0

## Di chuyển file dataset.zip

In [ ]:

import os
import shutil

# Đường dẫn thư mục chứa file ZIP
source_dir = '/kaggle/working/data'  # Thay thế bằng đường dẫn thực tế

# Tên file ZIP
zip_file_name = 'dataset.zip'

# Đường dẫn đầy đủ đến file ZIP trong thư mục nguồn
source_file_path = os.path.join(source_dir, zip_file_name)

# Đường dẫn đích (thư mục /kaggle/working/dataset)
destination_dir = '/kaggle/working/'

# Tạo thư mục đích nếu nó chưa tồn tại
os.makedirs(destination_dir, exist_ok=True)

# Đường dẫn đầy đủ đến file ZIP trong thư mục đích
destination_file_path = os.path.join(destination_dir, zip_file_name)

# Di chuyển file ZIP
try:
    shutil.move(source_file_path, destination_file_path)
    print(f"File '{zip_file_name}' đã được di chuyển từ '{source_dir}' đến '{destination_dir}' thành công!")
except FileNotFoundError:
    print(f"Lỗi: Không tìm thấy file '{zip_file_name}' trong thư mục '{source_dir}'.")
except Exception as e:
    print(f"Lỗi khi di chuyển file: {e}")

File 'dataset.zip' đã được di chuyển từ '/kaggle/working/data' đến '/kaggle/working/' thành công!


In [ ]:
import torch

# Kiểm tra và chọn thiết bị (GPU nếu có, nếu không dùng CPU)
device = "0" if torch.cuda.is_available() else "cpu"
print(f"Using device: {device}")

# Chạy YOLOv8 với thiết bị đã chọn
!yolo task=detect mode=train model=yolov8m.pt data=/kaggle/working/data/data.yaml epochs=90 imgsz=640 device={device} degrees=45


Using device: 0
Ultralytics 8.3.98 🚀 Python-3.10.12 torch-2.5.1+cu121 CUDA:0 (Tesla P100-PCIE-16GB, 16269MiB)
engine/trainer: task=detect, mode=train, model=yolov8m.pt, data=/kaggle/working/data/data.yaml, epochs=90, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=Tru

## 📊 Summary

- ⏱️ **Time**: `0.640 hours` (≈38.4 phút)  
- 🎯 **P (Precision - Độ chính xác)**: `0.878`  
- 📊 **R (Recall - Độ nhạy)**: `0.874`  
- 📈 **mAP50 (Mean Average Precision @0.5)**: `0.935`  
- 📉 **mAP50-95 (Mean Average Precision @0.5:0.95)**: `0.460`

---

## ⚡ Inference Speed (Tốc độ suy luận)

- 🧹 **Preprocess**: `0.2ms`  
- 🧠 **Inference**: `10.5ms`  
- 🛠️ **Postprocess**: `2.0ms`

---

✅ **Model**: `YOLOv8m`  
✅ **Epochs**: `90`  
✅ **Images (val)**: `166`  
✅ **Instances (val)**: `297`  
✅ **Device**: `Tesla P100-PCIE-16GB (CUDA:0)`

📁 **Kết quả lưu tại**: `runs/detect/train`

In [ ]:
!zip -r train_CLI.zip /kaggle/working/runs/detect/train

updating: kaggle/working/runs/detect/train/ (stored 0%)
updating: kaggle/working/runs/detect/train/confusion_matrix_normalized.png (deflated 38%)
updating: kaggle/working/runs/detect/train/train_batch0.jpg (deflated 2%)
updating: kaggle/working/runs/detect/train/val_batch0_pred.jpg (deflated 7%)
updating: kaggle/working/runs/detect/train/val_batch1_labels.jpg (deflated 9%)
updating: kaggle/working/runs/detect/train/train_batch1.jpg (deflated 2%)
updating: kaggle/working/runs/detect/train/args.yaml (deflated 53%)
updating: kaggle/working/runs/detect/train/train_batch2.jpg (deflated 2%)
updating: kaggle/working/runs/detect/train/train_batch2962.jpg (deflated 7%)
updating: kaggle/working/runs/detect/train/val_batch1_pred.jpg (deflated 8%)
updating: kaggle/working/runs/detect/train/R_curve.png (deflated 17%)
updating: kaggle/working/runs/detect/train/results.csv (deflated 60%)
updating: kaggle/working/runs/detect/train/val_batch2_labels.jpg (deflated 8%)
updating: kaggle/working/runs/detec

In [ ]:
import torch
from ultralytics import YOLO

# Kiểm tra và chọn thiết bị (GPU nếu có, nếu không thì dùng CPU)
device = 0 if torch.cuda.is_available() else "cpu"
print(f"Using device: {device}")

# Load mô hình YOLOv8m
model = YOLO("yolov8m.pt")

# Huấn luyện mô hình với tập dữ liệu của bạn
model.train(
    data="/kaggle/working/data/data.yaml",
    epochs=90,
    imgsz=640,
    device=device,
    degrees=45
)


Using device: 0
Ultralytics 8.3.98 🚀 Python-3.10.12 torch-2.5.1+cu121 CUDA:0 (Tesla P100-PCIE-16GB, 16269MiB)
engine/trainer: task=detect, mode=train, model=yolov8m.pt, data=/kaggle/working/data/data.yaml, epochs=90, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=8, project=None, name=train2, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=Tr

train: Scanning /kaggle/working/data/train/labels.cache... 580 images, 0 backgrounds, 0 corrupt: 100%|██████████| 580/580 [00:00<?, ?it/s]

albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))



val: Scanning /kaggle/working/data/valid/labels.cache... 166 images, 0 backgrounds, 0 corrupt: 100%|██████████| 166/166 [00:00<?, ?it/s]


Plotting labels to runs/detect/train2/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 77 weight(decay=0.0), 84 weight(decay=0.0005), 83 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 640 train, 640 val
Using 4 dataloader workers
Logging results to runs/detect/train2
Starting training for 90 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/90      8.53G      1.953      2.302      2.163          7        640: 100%|██████████| 37/37 [00:23<00:00,  1.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297     0.0142      0.391    0.00923    0.00287



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/90      8.53G      1.917      2.009      2.064         14        640: 100%|██████████| 37/37 [00:22<00:00,  1.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.59it/s]

                   all        166        297    0.00688      0.212    0.00363   0.000917



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/90      8.53G      1.946      1.924      2.115         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.00it/s]

                   all        166        297   2.01e-05    0.00337   1.08e-05   1.08e-06



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/90      8.53G      1.921      1.993      2.037         13        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.73it/s]

                   all        166        297     0.0131     0.0101    0.00109   0.000241



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/90      8.53G      1.862      1.879      2.012         21        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297     0.0186       0.38     0.0118    0.00321



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/90      8.53G      1.798      1.797      1.969         13        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297     0.0183       0.33     0.0105     0.0033



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/90      8.53G      1.809      1.753      1.969         19        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.68it/s]

                   all        166        297      0.018      0.781     0.0168    0.00635



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/90      8.53G      1.748      1.736      1.916          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.68it/s]

                   all        166        297      0.516      0.579       0.51      0.159



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/90      8.53G      1.747      1.768      1.971         13        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.602      0.431      0.432      0.134



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/90      8.53G      1.699      1.653      1.877          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.74it/s]

                   all        166        297      0.238      0.101     0.0662     0.0165



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/90      8.53G      1.701      1.645       1.88         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.68it/s]

                   all        166        297      0.569      0.657      0.603      0.212



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/90      8.53G      1.688       1.73      1.866          4        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297       0.41      0.744      0.477      0.158



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/90      8.53G      1.625      1.626      1.812          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.67it/s]

                   all        166        297      0.539       0.66      0.502      0.151



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/90      8.53G      1.594      1.621      1.833         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.676      0.721      0.645       0.22



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/90      8.53G      1.633      1.562      1.838         11        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.664      0.579      0.603      0.205



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/90      8.53G        1.6      1.597       1.77         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.742      0.746       0.78      0.266



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/90      8.53G      1.634      1.596      1.812          4        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.628      0.744      0.627      0.202



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/90      8.53G      1.618      1.509      1.781         17        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.66it/s]

                   all        166        297      0.692      0.727      0.702       0.24



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/90      8.53G      1.617      1.545      1.817         14        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.539      0.596      0.541      0.189



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/90      8.53G      1.598      1.559      1.804         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.636      0.717      0.656      0.234



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/90      8.53G      1.576      1.493      1.796         11        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.68it/s]

                   all        166        297       0.52      0.552      0.423      0.151



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/90      8.53G      1.589      1.501      1.778         14        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.632      0.671      0.601      0.208



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/90      8.53G      1.516      1.476      1.745         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.729      0.737      0.723      0.256



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/90      8.53G       1.54      1.457      1.789          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.704      0.781      0.754      0.279



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/90      8.53G      1.546      1.454      1.748          7        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.537      0.677      0.529      0.178



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/90      8.53G      1.527      1.497      1.743          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.533      0.727       0.56       0.19



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/90      8.53G      1.505      1.437      1.717         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.634       0.73      0.663      0.238



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/90      8.53G      1.519      1.456      1.726         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.662      0.751      0.699      0.229



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/90      8.53G      1.513      1.482      1.713          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.618      0.692      0.663      0.229



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/90      8.53G      1.503      1.474      1.684         13        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.705      0.773      0.732      0.281



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/90      8.53G      1.501      1.411      1.709         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.662      0.727       0.71      0.237



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/90      8.53G      1.479      1.388      1.686         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.682      0.663      0.662      0.255



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/90      8.53G      1.444       1.42       1.68         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.738      0.694      0.714      0.278



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/90      8.53G      1.454      1.414      1.702         21        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.739      0.754      0.738      0.258



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/90      8.53G      1.506       1.45      1.733         19        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.613      0.687      0.673      0.215



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/90      8.53G      1.463      1.372      1.678         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.667      0.761      0.699      0.252



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/90      8.53G      1.496      1.403      1.719         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.735      0.815      0.829      0.321



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/90      8.53G      1.455      1.375      1.692         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.842      0.878      0.896      0.399



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/90      8.53G      1.473      1.381      1.699         16        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.754      0.781      0.819      0.334



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/90      8.53G      1.458      1.385      1.681          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.787      0.832      0.832      0.345



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/90      8.53G      1.475      1.363      1.682         28        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.666      0.812       0.77        0.3



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/90      8.53G       1.44       1.39      1.664         14        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.801      0.781      0.833      0.339



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/90      8.53G      1.426      1.338      1.672          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.838      0.865      0.907      0.385



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/90      8.53G      1.426      1.361      1.666         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.788      0.865       0.85      0.341



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/90      8.53G      1.418      1.287      1.624          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.852      0.865      0.901      0.375



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/90      8.53G      1.428      1.311      1.621         17        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.791      0.811      0.864      0.367



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/90      8.53G      1.382      1.339      1.626          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.847      0.819      0.881      0.413



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/90      8.53G      1.434      1.384      1.683          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.847      0.855      0.884      0.395



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/90      8.53G      1.412      1.289      1.639         19        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.68it/s]

                   all        166        297      0.846      0.838      0.888       0.35



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/90      8.53G       1.37      1.309      1.635          8        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.825       0.86      0.882      0.353



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      51/90      8.53G      1.396      1.338      1.639         11        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.823      0.816      0.888      0.375



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      52/90      8.53G      1.375      1.308      1.633         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.813      0.828      0.868      0.388



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      53/90      8.53G       1.39      1.285      1.636         18        640: 100%|██████████| 37/37 [00:22<00:00,  1.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.853      0.845      0.905      0.413



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      54/90      8.53G      1.392      1.294      1.631         28        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297       0.86      0.868      0.907      0.395



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      55/90      8.53G      1.397      1.287      1.616         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.828      0.896      0.897      0.381



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      56/90      8.53G      1.355      1.245      1.596         11        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.884      0.872      0.929      0.389



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      57/90      8.53G      1.379      1.255      1.627          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.892      0.845      0.924      0.416



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      58/90      8.53G      1.352      1.254        1.6         34        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297       0.89      0.875      0.933      0.405



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      59/90      8.53G      1.383       1.28      1.614         37        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.869      0.871      0.916      0.408



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      60/90      8.53G      1.374      1.283      1.612         11        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.821      0.862      0.892      0.389



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      61/90      8.53G      1.336      1.264      1.582         16        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.874      0.859      0.915      0.407



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      62/90      8.53G      1.341      1.252      1.592         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297       0.86      0.892      0.922      0.442



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      63/90      8.53G      1.338      1.259      1.609         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.839      0.896      0.921      0.413



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      64/90      8.53G       1.34      1.262      1.592         23        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.856      0.899      0.898      0.392



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      65/90      8.53G      1.333      1.234      1.582          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.927      0.902      0.932      0.417



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      66/90      8.53G      1.337      1.265      1.605          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.881      0.872      0.919      0.437



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      67/90      8.53G      1.299      1.226      1.575         18        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.881      0.872       0.92      0.449



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      68/90      8.53G      1.317      1.217      1.567          9        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.871      0.885      0.925      0.436



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      69/90      8.53G        1.3      1.222      1.561         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.876      0.878      0.918      0.403



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      70/90      8.53G      1.301       1.23      1.561         10        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.65it/s]

                   all        166        297       0.88      0.866       0.92      0.445



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      71/90      8.53G      1.292      1.197      1.548         15        640: 100%|██████████| 37/37 [00:22<00:00,  1.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.864      0.879      0.923      0.444



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      72/90      8.53G      1.295      1.192      1.558         21        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.876      0.906      0.924      0.401



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      73/90      8.53G      1.293      1.176      1.545         21        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.847      0.841      0.899      0.406



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      74/90      8.53G      1.274      1.158      1.559         13        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.73it/s]

                   all        166        297        0.9      0.859      0.934      0.445



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      75/90      8.53G      1.313      1.154      1.566          6        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.881      0.872      0.935       0.46



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      76/90      8.53G      1.281      1.173      1.552         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.857      0.847      0.916       0.45



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      77/90      8.53G       1.29      1.119      1.553         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.70it/s]

                   all        166        297      0.846      0.872      0.907      0.432



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      78/90      8.53G      1.282      1.153      1.567         11        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297       0.86      0.869      0.925      0.427



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      79/90      8.53G      1.293      1.152      1.569         14        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.861      0.882      0.932      0.446



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      80/90      8.53G      1.266      1.151      1.557         17        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.872      0.847      0.923      0.455


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      81/90      8.53G      1.193      1.009      1.549          5        640: 100%|██████████| 37/37 [00:23<00:00,  1.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297       0.88      0.869      0.923      0.437



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      82/90      8.53G      1.188     0.9749      1.547          5        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.69it/s]

                   all        166        297      0.874      0.879      0.938      0.441



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      83/90      8.53G      1.195     0.9824      1.557          5        640: 100%|██████████| 37/37 [00:22<00:00,  1.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.887      0.901       0.93      0.417



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      84/90      8.53G      1.183     0.9432      1.544         30        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.896      0.886      0.936      0.431



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      85/90      8.53G      1.155     0.9482      1.514          6        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.873      0.862      0.913      0.429



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      86/90      8.53G      1.147     0.9486      1.513          7        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.865      0.886      0.927      0.433



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      87/90      8.53G      1.156     0.9391      1.519          4        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297      0.852      0.869      0.919      0.431



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      88/90      8.53G      1.146     0.9395       1.52          4        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.71it/s]

                   all        166        297       0.88      0.864      0.922      0.446



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      89/90      8.53G      1.143     0.9282      1.511         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.73it/s]

                   all        166        297      0.875      0.869      0.924       0.45



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      90/90      8.53G       1.16      0.929      1.522         12        640: 100%|██████████| 37/37 [00:22<00:00,  1.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.72it/s]

                   all        166        297      0.877      0.865      0.926      0.453



90 epochs completed in 0.642 hours.
Optimizer stripped from runs/detect/train2/weights/last.pt, 52.0MB
Optimizer stripped from runs/detect/train2/weights/best.pt, 52.0MB

Validating runs/detect/train2/weights/best.pt...
Ultralytics 8.3.98 🚀 Python-3.10.12 torch-2.5.1+cu121 CUDA:0 (Tesla P100-PCIE-16GB, 16269MiB)
Model summary (fused): 92 layers, 25,840,339 parameters, 0 gradients, 78.7 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.94it/s]


                   all        166        297      0.878      0.874      0.935       0.46


/usr/local/lib/python3.10/dist-packages/matplotlib/colors.py:721: RuntimeWarning: invalid value encountered in less
  xa[xa < 0] = -1
/usr/local/lib/python3.10/dist-packages/matplotlib/colors.py:721: RuntimeWarning: invalid value encountered in less
  xa[xa < 0] = -1


Speed: 0.2ms preprocess, 10.6ms inference, 0.0ms loss, 2.9ms postprocess per image
Results saved to runs/detect/train2


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7c2bdf07b2e0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.048048, 

# 🔍 Kết quả huấn luyện YOLOv8

| Thông số               | Giá trị       |
|------------------------|---------------|
| ⏱️ Thời gian huấn luyện   | 0.642 giờ       |
| 🎯 Precision (P)           | 0.878         |
| 📊 Recall (R)              | 0.874         |
| 📈 mAP@0.5                 | 0.935         |
| 📉 mAP@0.5:0.95            | 0.460         |

# ⚡ Tốc độ suy luận (Inference Speed)

| Giai đoạn       | Thời gian (ms) |
|------------------|----------------|
| Preprocess       | 0.2 ms         |
| Inference        | 10.6 ms        |
| Postprocess      | 2.9 ms         |


In [ ]:
!zip -r train_API.zip /kaggle/working/runs/detect/train2

  adding: kaggle/working/runs/detect/train2/ (stored 0%)
  adding: kaggle/working/runs/detect/train2/confusion_matrix_normalized.png (deflated 38%)
  adding: kaggle/working/runs/detect/train2/train_batch0.jpg (deflated 2%)
  adding: kaggle/working/runs/detect/train2/val_batch0_pred.jpg (deflated 7%)
  adding: kaggle/working/runs/detect/train2/val_batch1_labels.jpg (deflated 9%)
  adding: kaggle/working/runs/detect/train2/train_batch1.jpg (deflated 2%)
  adding: kaggle/working/runs/detect/train2/args.yaml (deflated 52%)
  adding: kaggle/working/runs/detect/train2/train_batch2.jpg (deflated 2%)
  adding: kaggle/working/runs/detect/train2/train_batch2962.jpg (deflated 7%)
  adding: kaggle/working/runs/detect/train2/val_batch1_pred.jpg (deflated 8%)
  adding: kaggle/working/runs/detect/train2/R_curve.png (deflated 17%)
  adding: kaggle/working/runs/detect/train2/results.csv (deflated 60%)
  adding: kaggle/working/runs/detect/train2/val_batch2_labels.jpg (deflated 8%)
  adding: kaggle/worki